x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_enrollment_flyout/advanced_agent_authentication_settings.tsx

```diff
@@ -0,0 +1,194 @@
+        enrollmentAPIKeys[0].policy_id === agentPolicyId
+      ) {
+        const enrollmentAPIKeyId = enrollmentAPIKeys[0].id;
+        setSelectedEnrollmentApiKey(enrollmentAPIKeyId);
+      }
+    },
+    [enrollmentAPIKeys, selectedEnrollmentApiKey, agentPolicyId]
+  );
+  return (
+    <>
+      <EuiButtonEmpty
+        flush="left"
+        iconType={isAuthenticationSettingsOpen ? 'arrowDown' : 'arrowRight'}
+        onClick={() => setIsAuthenticationSettingsOpen(!isAuthenticationSettingsOpen)}
+      >
+        <FormattedMessage
+          id="xpack.fleet.enrollmentStepAgentPolicy.showAuthenticationSettingsButton"
+          defaultMessage="Authentication settings"
+        />
+      </EuiButtonEmpty>
+      {isAuthenticationSettingsOpen && (
+        <>
+          <EuiSpacer size="m" />
+          {enrollmentAPIKeys.length && selectedEnrollmentApiKey ? (
+            <EuiSelect
+              fullWidth
+              options={enrollmentAPIKeys.map((key) => ({
+                value: key.id,
+                text: key.name,
+              }))}
+              value={selectedEnrollmentApiKey || undefined}
+              prepend={
+                <EuiText>
+                  <FormattedMessage
+                    id="xpack.fleet.enrollmentStepAgentPolicy.enrollmentTokenSelectLabel"
+                    defaultMessage="Enrollment token"
+                  />
+                </EuiText>
+              }
+              onChange={(e) => {
+                setSelectedEnrollmentApiKey(e.target.value);
+              }}
+            />
+          ) : (
+            <EuiCallOut
+              color="warning"
+              title={i18n.translate(
+                'xpack.fleet.enrollmentStepAgentPolicy.noEnrollmentTokensForSelectedPolicyCallout',
+                {
+                  defaultMessage: 'There are no enrollment tokens for the selected agent policy',
+                }
+              )}
+            >
+              <div className="eui-textBreakWord">
+                <FormattedMessage
+                  id="xpack.fleet.agentEnrenrollmentStepAgentPolicyollment.noEnrollmentTokensForSelectedPolicyCalloutDescription"
+                  defaultMessage="You must create and enrollment token in order to enroll agents with this policy"
+                />
+              </div>
+              <EuiSpacer size="m" />
+              <EuiButton
+                iconType="plusInCircle"
+                isLoading={isLoadingEnrollmentKey}
+                fill
+                onClick={() => {
+                  setIsLoadingEnrollmentKey(true);
+                  if (agentPolicyId) {
+                    sendCreateEnrollmentAPIKey({ policy_id: agentPolicyId })
+                      .then((res) => {
+                        if (res.error) {
+                          throw res.error;
+                        }
+                        setIsLoadingEnrollmentKey(false);
+                        if (res.data?.item) {
```

Comment:
Let's invert this condition and bail out early, since the body of this `if` is a few lines long.

```ts
if (!res.data?.item) {
  return
}

setEnrollmentAPIKeys ...
```